nvflare/client/api.py

```diff
@@ -49,46 +51,65 @@ def init(config: Union[str, Dict] = f"config/{CONFIG_EXCHANGE}", rank: Optional[
     else:
         raise ValueError("config should be either a string or dictionary.")
 
-    mdx = None
+    dx = None
     if rank == "0":
-        if client_config.get_exchange_format() == ModelExchangeFormat.PYTORCH:
+        if client_config.get_exchange_format() == ExchangeFormat.PYTORCH:
             tensor_decomposer, ok = optional_import(module="nvflare.app_opt.pt.decomposers", name="TensorDecomposer")
             if ok:
                 fobs.register(tensor_decomposer)
             else:
-                raise RuntimeError(f"Can't import TensorDecomposer for format: {ModelExchangeFormat.PYTORCH}")
+                raise RuntimeError(f"Can't import TensorDecomposer for format: {ExchangeFormat.PYTORCH}")
 
         # TODO: make Pipe configurable
-        mdx = FilePipeModelExchanger(data_exchange_path=client_config.get_exchange_path())
+        dx = FilePipeDataExchanger(
+            supported_topics=client_config.get_supported_topics(),
+            data_exchange_path=client_config.get_exchange_path(),
+            pipe_name=client_config.get_pipe_name(),
+        )
 
-    PROCESS_MODEL_REGISTRY[pid] = ModelRegistry(client_config, rank, mdx)
+    PROCESS_MODEL_REGISTRY[pid] = ModelRegistry(client_config, rank, dx)
 
 
-def _get_model_registry() -> Optional[ModelRegistry]:
+def _get_model_registry() -> ModelRegistry:
     pid = os.getpid()
     if pid not in PROCESS_MODEL_REGISTRY:
         raise RuntimeError("needs to call init method first")
     return PROCESS_MODEL_REGISTRY[pid]
 
 
-def receive() -> FLModel:
+def receive_task(timeout: Optional[float] = None) -> Tuple[str, Any, dict]:
+    """Receives a task from NVFlare side."""
+    model_registry = _get_model_registry()
+    task = model_registry.get_task(timeout)
+    if task is None:
+        return "", None, {}
+    return model_registry.task_name, task.data, task.meta
+
+
+def send_task_result(data: dict, meta: dict, return_code: str):
+    model_registry = _get_model_registry()
+    model_registry.send_task(data, meta, return_code)
+
+
+def receive(timeout: Optional[float] = None) -> Optional[FLModel]:
     """Receives model from NVFlare side.
 
     Returns:
         An FLModel received.
     """
     model_registry = _get_model_registry()
-    return model_registry.get_model()
+    return model_registry.get_model(timeout)
 
 
 def send(fl_model: FLModel, clear_registry: bool = True) -> None:
     """Sends the model to NVFlare side.
 
     Args:
+        fl_model (FLModel): Sends a FLModel object.
         clear_registry (bool): To clear the registry or not.
     """
     model_registry = _get_model_registry()
-    model_registry.send(model=fl_model)
+    model_registry.send_model(model=fl_model)
```

Comment:
similarly shouldn't you convert FLModel to task and then send call send_task()?
The layering isn't very clear: who is responsible for conversion btw ExchangeTask and FLModel?

Reply:
Introduce ModelRegistry that do this convenient conversion